ArcGISRuntimeSDKQt_CppSamples/Analysis/AnalyzeViewshed/AnalyzeViewshed.cpp

```diff
@@ -0,0 +1,223 @@
+  FeatureCollectionTable* inputFeatures = new FeatureCollectionTable(QList<Field>(),
+                                                                     GeometryType::Point,
+                                                                     SpatialReference::webMercator(),
+                                                                     this);
+
+
+  // Create a new feature from the feature collection table. It will not have a coordinate location (x,y) yet
+  Feature* inputFeature = inputFeatures->createFeature(this);
+
+  // Assign a physical location to the new point feature based upon where the user clicked on the map view
+  inputFeature->setGeometry(m_inputOverlay->graphics()->at(0)->geometry());
+
+  // connect to addFeature status changed signal
+  connect(inputFeatures, &FeatureCollectionTable::addFeatureCompleted, this, [this, inputFeatures](QUuid, bool success)
+  {
+    if (!success)
+      return;
+
+    // Create the parameters that are passed to the used geoprocessing task
+    GeoprocessingParameters viewshedParameters = GeoprocessingParameters(GeoprocessingExecutionType::SynchronousExecute);
+
+    // Request the output features to use the same SpatialReference as the map view
+    viewshedParameters.setOutputSpatialReference(SpatialReference::webMercator());
+
+    // Add an input location to the geoprocessing parameters
+    QMap<QString, GeoprocessingParameter*> inputs;
+    inputs["Input_Observation_Point"] = new GeoprocessingFeatures(inputFeatures, this);
+    viewshedParameters.setInputs(inputs);
+
+    // Create the job that handles the communication between the application and the geoprocessing task
+    GeoprocessingJob* viewshedJob = m_viewshedTask->createJob(viewshedParameters);
+
+    // Create signal handler for the job
+    connect(viewshedJob, &GeoprocessingJob::jobStatusChanged, this, [this, viewshedJob]()
+    {
+      switch (viewshedJob->jobStatus()) {
+          case JobStatus::Failed:
+            emit displayErrorDialog("Geoprocessing Task failed", !viewshedJob->error().isEmpty() ? viewshedJob->error().message() : "Unknown error.");
+            m_viewshedInProgress = false;
+            emit viewshedInProgressChanged();
+            m_jobStatus = "Job failed";
+            emit jobStatusChanged();
+            break;
+          case JobStatus::Started:
+            m_viewshedInProgress = true;
+            emit viewshedInProgressChanged();
+            m_jobStatus = "Job in progress...";
+            emit jobStatusChanged();
+            break;
+          case JobStatus::Paused:
+            m_viewshedInProgress = false;
+            emit viewshedInProgressChanged();
+            m_jobStatus = "Job paused...";
+            emit jobStatusChanged();
+            break;
+          case JobStatus::Succeeded:
+            m_viewshedInProgress = false;
+            emit viewshedInProgressChanged();
+            m_jobStatus = "Job succeeded";
+            emit jobStatusChanged();
```

Comment:
You can move 
```
emit viewshedInProgressChanged();
emit jobStatusChanged();
```

Out of the switch statement, unless there are order problems.